lang/cs/Org.Apache.REEF.Driver.Tests/TestTaskStates.cs

```diff
@@ -0,0 +1,152 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using Org.Apache.REEF.Driver.Task;
+using Xunit;
+
+namespace Org.Apache.REEF.Driver.Tests
+{
+    /// <summary>
+    /// The test cases in this classes test TaskState and transitions
+    /// </summary>
+    public class TestTaskStates
+    {
+        [Fact]
+        public void TestNewToCompleteSenario()
+        {
+            var taskState = new DriverTaskState();
+            Assert.True(taskState.CurrentState.Equals(TaskTransitionState.TaskNew));
+            Assert.False(taskState.IsFinalState(), "TaskNew is not final state.");
```

Comment:
Error message should be "TaskNew should not be a final state" (if Assert fails, then it is a final state, and error message says it is not, which is confusing). Same for the rest of `Assert.False(taskState.IsFinalState())` checks